torch/distributed/_sharded_tensor/ops/linear.py

```diff
@@ -0,0 +1,129 @@
+import torch
+import torch.distributed as dist
+from torch.distributed._sharding_spec import ChunkShardingSpec
+from torch.distributed._sharding_spec._internals import (
+    get_split_size,
+    get_chunked_dim_size,
+)
+from typing import List
+
+
+def sharded_linear(types, args, kwargs, pg):
+    from torch.distributed._sharded_tensor import ShardedTensor
+
+    input = args[0]
+    weight = args[1]
+    bias = kwargs["bias"]
+
+    # Validate types
+    if not isinstance(input, torch.Tensor) or not isinstance(bias, torch.Tensor):
+        raise TypeError("input and bias need to be torch.Tensor")
+    if not isinstance(weight, ShardedTensor):
+        raise TypeError("weight needs to be ShardedTensor")
+
+    if input.size()[1] != weight.size()[1]:
+        raise ValueError(
+            f'Input dim: {input.size()[1]} does not match '
+            f'appropriate weight dim: {weight.size()[1]}')
+    if not isinstance(weight._sharding_spec, ChunkShardingSpec):
+        raise ValueError("Only ChunkShardingSpec supported for ShardedTensor ops!")
+    if len(weight.local_shards()) != 1:
+        raise ValueError("Only one local shard supported!")
+
+    local_shard = weight.local_shards()[0].tensor
+    local_shard_t = local_shard.t().contiguous()
+    sharding_dim = weight._sharding_spec.dim
+    world_size = dist.get_world_size(pg)
+    rank = dist.get_rank(pg)
+
+    if sharding_dim == 1:
+        return _handle_row_wise_sharding(input, world_size, weight, rank, local_shard_t, bias, pg)
+    elif sharding_dim == 0:
+        return _handle_col_wise_sharding(input, world_size, weight, local_shard_t, bias, pg)
+    else:
+        raise RuntimeError(f'nn.Linear weight sharded on dim {sharding_dim} not supported!')
+
```

Comment:
Would it be easy to add some comments/examples to explain the overall algo to make it easier to understand? 

Basically an example input, weight, world_size (maybe 3?) and what happens for each step.